source/extensions/quic_listeners/quiche/platform/string_utils.cc

```diff
@@ -0,0 +1,105 @@
+#include "extensions/quic_listeners/quiche/platform/string_utils.h"
+
+// NOLINT(namespace-envoy)
+//
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <arpa/inet.h>
+#include <cstring>
+#include <string>
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/escaping.h"
+#include "absl/strings/str_format.h"
+#include "common/common/assert.h"
+
+namespace quiche {
+
+void Base64Encode(const uint8_t* data, size_t data_len, std::string* output) {
```

Comment:
Can you reuse https://github.com/envoyproxy/envoy/blob/master/source/common/common/base64.h instead of some of these implementations?

Reply:
The two variants of Base64::encode both adds padding('=') at the end of output, which is the same as the absl::Base64Escape function used here.

Do you think I should add a third Base64::encode function, which takes an "add_padding" argument, and use it here?

Reply:
Yep, that would be great. I'm happy to reuse things in absl and QUICHE in preference to Envoy's implementation, but when it comes to reimplementing things in the adapter layer, I rather use some existing code.

Reply:
Done!